src/v/container/interval_set.h

```diff
@@ -51,13 +54,27 @@ class interval_set {
      */
     [[nodiscard]] std::pair<const_iterator, bool> insert(interval interval);
 
+    /**
+     * Exposed to make this type serde-able.
+     */
+    std::pair<const_iterator, bool> emplace(key_type k, mapped_type m) {
+        return set_.emplace(std::move(k), std::move(m));
```

Comment:
This can break container invariants if used outside of serde. Let's preserve the semantics and just call insert? These are only integral types so emplace doesn't buy much.


Maybe instead of doing this we should just add serde write and serde read static methods that work on the underlying container?